packages/std/src/memory.rs

```diff
@@ -49,24 +47,29 @@ pub fn release_buffer(buffer: Vec<u8>) -> *mut c_void {
 ///
 /// # Safety
 ///
-/// If ptr is non-nil, it must refer to a valid Region, which was previously returned by alloc,
+/// The ptr must refer to a valid Region, which was previously returned by alloc,
 /// and not yet deallocated. This call will deallocate the Region and return an owner vector
 /// to the caller containing the referenced data.
 ///
 /// Naturally, calling this function twice on the same pointer will double deallocate data
 /// and lead to a crash. Make sure to call it exactly once (either consuming the input in
 /// the wasm code OR deallocating the buffer from the caller).
-pub unsafe fn consume_region(ptr: *mut c_void) -> StdResult<Vec<u8>> {
+pub unsafe fn consume_region(ptr: *mut c_void) -> Vec<u8> {
     if ptr.is_null() {
-        return Err(null_pointer());
+        panic!("Region pointer is null");
     }
```

Comment:
I'm thinking, could this be replaced with:
```rust
assert!(!ptr.is_null(), "Region pointer is null");
````
and similarly for `region_start` below?

Reply:
I thought about this as well but then was not sure if assert is optimized out in production builds – what you typically have in C++

Reply:
Rust distinguishes [assert!](https://doc.rust-lang.org/std/macro.assert.html) from [debug_assert!](https://doc.rust-lang.org/std/macro.debug_assert.html). I initially commented that you might want to use those instead, before i realized it's not a good idea in this case.